File: frontend/bff/src/application/organizations/commands/update-industrial-profile.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface UpdateIndustrialProfileCommand {
  context: RequestContext;
  millingCapacity: number | null;
  mixMinPct: number | null;
  mixMaxPct: number | null;
  mixGuidancePct: number | null;
}

File: backend/core-service/src/Fix.Application/UseCases/Organizations/Commands/UpdateIndustrialProfileCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Organizations.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

[RequireRole(RoleCodes.EditOrganization)]
public sealed record UpdateIndustrialProfileCommand(
    Guid UserId,
    Guid OrganizationId,
    decimal? MillingCapacity,
    decimal? MixMinPct,
    decimal? MixMaxPct,
    decimal? MixGuidancePct)
    : ICommand<OrganizationSetupDto>, IOrganizationRequest;

